rust-mode-tests.el

```diff
@@ -3618,11 +3619,13 @@ let b = 1;"
                 ("file6.rs" "132" "34" compilation-error "file6.rs:132:34"))
                (("file5.rs" "12" "34" compilation-info "file5.rs:12:34"))
                ((like-previous-one "82" back-to-indentation compilation-info "82")
-                (like-previous-one "132" back-to-indentation compilation-info "132")))
+                (like-previous-one "132" back-to-indentation compilation-info "132"))
+               (("src/file7.rs" "12" "34" nil "src/file7.rs:12:34")))
```

Comment:
I've added a test now, but I'm not sure why this is `nil` and not `compilation-error`. Maybe someone more comfortable with Elisp can assist me.

Reply:
I am not very familiar with elisp either, so I could be wrong.
Here's what I think is happening here:
compilation-error is set
```
(defvar compilation-error "error"
  "Stem of message to print when no matches are found.")
```
And so an expression like `("file6.rs" "132" "34" compilation-error "file6.rs:132:34")` is equivalent to `("file6.rs" "132" "34" error "file6.rs:132:34")'`

That translates to `type` expression from the `compilation-error-regexp-alist` help doc. I'm getting that from this line
```
Each elt has the form (REGEXP FILE [LINE COLUMN TYPE HYPERLINK
HIGHLIGHT...]).  If REGEXP matches, the FILE’th subexpression
gives the file name, and the LINE’th subexpression gives the line
number.  The COLUMN’th subexpression gives the column number on
that line.
```
The definition of our regex is
```
defvar rustc-panics-compilation-regexps
   (let ((re (concat "thread '[^']+' panicked at " rustc-compilation-location)))
     (cons re '(2 3 4 nil 1)))
   "Specifications for matching panics in rustc invocations.
See `compilation-error-regexp-alist' for help on their format.")
```
And so we're capturing the:
2nd capture group for the file name
3rd capture group for the line number,
4th capture group for the column number,
nil for the type
1st capture group for the hyperlink.

We don't have an `error` or `warning` in our error message, like the others do, to parse out for the `type`, so our `type` is `nil`.

Reply:
so I think `nil` is correct. That, or invent a new type like `panic`. I'm not really sure.